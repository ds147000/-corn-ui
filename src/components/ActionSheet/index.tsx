/* eslint-disable import/export */

import ActionSheet  from './actionSheet'

export { ActionSheetItem } from './item'
export default ActionSheet
