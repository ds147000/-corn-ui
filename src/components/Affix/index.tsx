/* eslint-disable no-unreachable */
/* eslint-disable no-magic-numbers */
// #if _APP === 'weapp'
import Taro, { usePageScroll } from '@tarojs/taro'
// #endif
import React, { useCallback, useEffect, useMemo, useRef, useState } from 'react'
import { View } from '@tarojs/components'
import ClassNames from 'classnames'
import { ViewProps } from '../../types/View'
import { judge } from './utils/judge'

export interface AffixProps extends ViewProps {
  position?: 'top' | 'bottom' | 'right' | 'left'
  onChange?(fixed: boolean): void
}

const Affix: React.FC<AffixProps> = ({ children, position = 'top', onChange, className, ...props }) => {
  const [ fixed, setFixed ] = useState(false)
  const [ boxStyle, setBoxStyle ] = useState<React.CSSProperties>({})
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const ref = useRef<any>()

  const oldFixed = useRef<boolean>(false)

  const _class = useMemo(() => {
    return ClassNames('corn-affix', fixed && `corn-affix-${position}`, className)
  }, [ fixed, position, className ])

  const changeFixed = useCallback((value: boolean): void => {
    setFixed((state) => {
      oldFixed.current = state
      return value
    })
  }, [])

  // #if _APP === 'weapp'
  const system = useRef(Taro.getSystemInfoSync())
  const timer = useRef<NodeJS.Timeout>()

  const checkFlexd = (): void => {
    const node = ref.current

    const rootRect = {
      top: 0,
      left: 0,
      bottom: system.current.windowHeight,
      right: system.current.windowWidth
    } as DOMRectReadOnly

    Taro.createSelectorQuery()
      .select('#' + node.uid)
      .fields({ size: true, rect: true }, (res) => {
        if (!res) return

        setBoxStyle({ height: res?.height, width: res?.width })
        const result = judge(res as unknown as DOMRect, rootRect, position, 0)
        changeFixed(result)
      })
      .exec()
  }

  usePageScroll(() => {
    if (timer.current) clearTimeout(timer.current)
    timer.current = setTimeout(() => checkFlexd(), 10)
  })

  useEffect(() => {
    return (): void => {
      if (timer.current) clearTimeout(timer.current)
    }
  }, [])

  // #else
  useEffect(() => {
    // eslint-disable-next-line no-unreachable
    const el: HTMLDivElement = ref.current
    const rootBounds = {
      x: 0,
      y: 0,
      width: window.innerWidth,
      height: window.innerHeight,
      top: 0,
      left: 0,
      bottom: window.innerHeight,
      right: window.innerWidth
    } as DOMRectReadOnly

    setBoxStyle({ height: el.offsetHeight, width: el.offsetWidth })

    function handleScroll(): void {
      const item = el.getBoundingClientRect()
      const result = judge(item, rootBounds, position, 0)
      changeFixed(result)
    }
    handleScroll()
    window.addEventListener('scroll', handleScroll)
    return (): void => {
      window.removeEventListener('scroll', handleScroll)
    }
  }, [ position, changeFixed ])
  // #endif

  useEffect(() => {
    if (oldFixed.current !== fixed) {
      oldFixed.current = fixed
      onChange?.(fixed)
    }
  }, [ fixed, onChange ])

  return (
    <View style={boxStyle} ref={ref}>
      <View className={_class} {...props} >
        {children}
      </View>
    </View>
  )
}

export default Affix
