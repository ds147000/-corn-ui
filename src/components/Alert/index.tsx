import React from 'react'
import { View } from '@tarojs/components'
import classNames from 'classnames'
import { ViewProps } from '../../types/View'
import Icon from '../Icon'

const Alter: React.FC<ViewProps> = ({ children, className, ...props }) => {
  const _class = classNames(className, 'corn-alert', 'corn-f')
  return (
    <View className={_class} {...props} >
      <Icon name="error" className="icon" />
      {children}
    </View>
  )
}

export default Alter
