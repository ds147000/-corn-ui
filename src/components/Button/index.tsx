import React, { useMemo } from 'react'
import ClassNames from 'classnames'
import { Button as TaroButton } from '@tarojs/components'
import Link, { LinkProps } from '../Link'
import { ITouchEvent } from '../../types'

/** 按钮大小，max: 最大， big: 超大，large：大，middle：中，small：小，mini */
export type ButtonSize = 'max' | 'big' | 'large' | 'middle' | 'small' | 'mini'
export type ButtonType = 'primary' | 'warn' | 'error' | 'link' | 'pop' | 'default' | 'light' | 'stop'

export interface ButtonProps extends LinkProps {
  /** 按钮大小，max: 最大， big: 超大，large：大，middle：中，small：小，mini */
  size?: ButtonSize
  /** 按钮风格类型 */
  type?: ButtonType
  /** 图标 */
  icon?: React.ReactNode
  /** 将按钮宽度调整为其父宽度的选项	 */
  block?: boolean
  /** 镂空风格 */
  ghost?: boolean
  /** 是否禁用 */
  disabled?: boolean
  /** 跳转链接，填写启用A链接行为 */
  to?: string
  /** 表单点击行为 */
  formType?: 'submit' | 'reset'
  /** 开启固定宽度 */
  auto?: boolean
  /** 点击事件 */
  onClick?(e: ITouchEvent): void
}

const Button: React.FC<ButtonProps> = ({
  size = 'middle',
  type = 'primary',
  block,
  ghost,
  disabled,
  to,
  onClick,
  icon,
  children,
  className,
  formType,
  auto,
  ...props
}) => {
  const isIcon = icon !== undefined
  const isChildren = children !== undefined

  const _class = useMemo(() => {
    return ClassNames(
      'corn-btn',
      'corn-if corn-ac corn-jc',
      `corn-btn-${size}`,
      `corn-btn-${type}`,
      {
        'corn-btn-block': block,
        'corn-btn-ghost': ghost,
        'corn-btn-icon': isIcon && !isChildren,
        'corn-btn-child-icon': isIcon && isChildren,
        'corn-btn-disabled': disabled,
        'corn-btn-auto': auto
      },
      className
    )
  }, [ size, type, block, ghost, disabled, className, auto, isIcon, isChildren ])

  const _onClick = (e: ITouchEvent): void => {
    if (disabled) return

    onClick?.(e)
  }



  if (to) {
    return (
      <Link {...props} target="Text" to={to} className={_class} onClick={_onClick} disabled={disabled} type="default" >
        {children}
        {icon}
      </Link>
    )
  }

  return (
    <TaroButton {...props} formType={formType} className={_class} onClick={_onClick} >
      {children}
      {icon}
    </TaroButton>
  )
}

export default Button
