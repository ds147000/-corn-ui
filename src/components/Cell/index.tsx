import React, { useCallback } from 'react'
import ClassNames from 'classnames'
import { View, Text } from '@tarojs/components'
import { ViewProps } from '@tarojs/components/types/View'
import Icon from '../Icon'
import Link, { LinkProps } from '../Link'
import CellList from './list'

export interface CellProps extends ViewProps, LinkProps {
  /** 标题 */
  label?: string | React.ReactNode
  /** 内容 */
  value?: string | React.ReactNode
  /** 内容为空占位提示 */
  placeholder?: string | React.ReactNode
  /** 是否显示箭头 */
  arrow?: boolean
  /** 尾缀内容 */
  suffix?: React.ReactNode | string
  /** 禁用 */
  disabled?: boolean
  /** 跳转链接，填写启用A链接行为。同时会显示箭头 */
  to?: string
}

type CellView =  { List: typeof CellList }



const Cell: React.FC<CellProps> & CellView = ({
  label, value, arrow, suffix,
  disabled, to, children, onClick,
  className, placeholder, ...props
}) => {

  const _onClick = useCallback((event): void => {
    if (disabled) return
    onClick?.(event)
  }, [ onClick, disabled ])

  const _props: ViewProps = {
    className: ClassNames('xrk-cell xrk-f xrk-ac', disabled && 'xrk-cell-disable', className),
    onClick: _onClick,
    ...props
  }

  const showArrow = (arrow || Boolean(to)) && arrow !== false
  const showPlaceholder = Boolean(value) === false && Boolean(children) === false

  const content = (
    <>
      <View className="xrk-cell-label">{label}</View>
      <View className="xrk-cell-value">
        {value}{children}
        {showPlaceholder && <Text className="xrk-cell-placeholder">{placeholder}</Text>}
      </View>
      <View className="xrk-cell-suffix xrk-f xrk-ac">
        {suffix}
        {showArrow && <Icon name="big-right" data-testid="arrow" />}
      </View>
    </>
  )

  if (to)
    return (<Link {..._props} to={to} disabled={disabled} target="View" >{content}</Link>)

  return (<View {..._props} >{content}</View>)
}

Cell.List = CellList

export default Cell
