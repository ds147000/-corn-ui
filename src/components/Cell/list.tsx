import React, { useMemo } from 'react'
import ClassNames from 'classnames'
import { View } from '@tarojs/components'
import { ViewProps } from '../../types/View'

export interface CellListProps extends ViewProps {
  /** 是否显示边框 */
  line?: boolean
}

const CellList: React.FC<CellListProps> = ({ children, line, className, ...props }) => {
  const _class = useMemo(() => {
    return ClassNames(line && 'corn-cell-list-line', className)
  }, [ line, className ])

  return (
    <View {...props} className={_class}>
      {children}
    </View>
  )
}

export default CellList
