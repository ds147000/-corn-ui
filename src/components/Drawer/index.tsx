import React, { useCallback, useEffect, useMemo, useState } from 'react'
import classNames from 'classnames'
import { View } from '@tarojs/components'

type AnimateStatus = 'show' | 'hide' | 'over'

export interface DrawerProps {
  /** 是否显示 */
  visible: boolean
  /** 弹出的方向，可以设置左右上下中间, 默认 bottom */
  position?: 'left' | 'right' | 'bottom' | 'top' | 'center'
  /** 是否显示遮罩层 */
  mask?: boolean
  /** 点击蒙层是否允许关闭 */
  maskClosable?: boolean
  /** 关闭回调 */
  onClose?(): void
  /** 完全隐藏回调 */
  onHide?(): void
}

const Drawer: React.FC<DrawerProps> = ({
  visible,
  position = 'bottom',
  mask = true,
  maskClosable = true,
  onClose,
  onHide,
  children
}) => {
  const [ status, setStatus ] = useState<AnimateStatus>('over')
  const _class = useMemo(() => {
    return classNames(
      'xrk-drawer-body',
      status !== 'over' && `xrk-drawer-${position}-${status}`,
      `xrk-drawer-${position}`
    )
  }, [ position, status ])

  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const onAnimationEnd = useCallback((event: any) => {
    const animationName: string = event.animationName ||
      event.detail?.animationName ||
      event.nativeEvent?.animationName ||
      event.target.className

    // eslint-disable-next-line no-magic-numbers
    if (animationName.toLocaleUpperCase().indexOf('HIDE') !== -1) setStatus('over')
  }, [])

  const onClickMask = (): void => {
    if (maskClosable) onClose?.()
  }

  useEffect(() => {
    if (visible === true && status !== 'show')
      setStatus('show')

    else if (visible === false && status === 'show')
      setStatus('hide')

  }, [ visible, status ])

  if (visible === false && status === 'over') {
    onHide?.()
    return null
  }

  return (
    <>
      {mask && (
        <View
          className={`xrk-drawer-mask xrk-drawer-mask-${status}`}
          data-testid="mask"
          onClick={onClickMask}
        />
      )}
      <View className={_class} onAnimationEnd={onAnimationEnd} data-testid="body" >
        {children}
      </View>
    </>
  )
}

export default Drawer
