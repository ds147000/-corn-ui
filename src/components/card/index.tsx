/*
 * @Description:
 * @Author: zhoulong.yang
 * @Date: 2021-06-07 12:03:34
 * @LastEditors: zhoulong.yang
 * @LastEditTime: 2021-06-07 15:43:55
 */
import React from 'react'
import { View, Text } from '@tarojs/components'

const Card: React.FC = () => {
  return (
    <View>
      <Text>卡片</Text>
    </View>
  )
}

export default Card
