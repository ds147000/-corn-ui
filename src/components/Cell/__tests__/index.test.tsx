import React from 'react'
import { fireEvent, render, waitFor } from '@testing-library/react'
import { BrowserRouter } from 'react-router-dom'
import { Text } from '@tarojs/components'
import Cell from '../index'

describe('Cell', () => {

  test('basis', () => {
    const screen = render(<Cell label="限时奖励" value="六月阶梯奖励活动" arrow />)
    screen.getByText('限时奖励')
    screen.getByText('六月阶梯奖励活动')
    screen.getByTestId('arrow')
    expect(screen.container).toMatchSnapshot()
  })


  test('children', () => {
    const screen = render(
      <Cell label="限时奖励" arrow >
        <Text>六月阶梯奖励活动</Text>
      </Cell>
    )
    screen.getByText('限时奖励')
    screen.getByText('六月阶梯奖励活动')
    screen.getByTestId('arrow')
    expect(screen.container).toMatchSnapshot()
  })

  test('click', () => {
    const click = jest.fn()
    const screen = render(<Cell label="限时奖励" value="六月阶梯奖励活动" arrow onClick={click} />)
    fireEvent.click(screen.getByText('限时奖励'))
    expect(click).toHaveBeenCalledTimes(1)
  })

  test('click 2 ', () => {
    const click = jest.fn()
    const screen = render(<Cell label="限时奖励" value="六月阶梯奖励活动" arrow onClick={click} />)
    fireEvent.click(screen.getByText('六月阶梯奖励活动'))
    expect(click).toHaveBeenCalledTimes(1)
  })

  test('click 3', () => {
    const click = jest.fn()
    const screen = render(<Cell label="限时奖励" value="六月阶梯奖励活动" arrow onClick={click} />)
    fireEvent.click(screen.getByTestId('arrow'))
    expect(click).toHaveBeenCalledTimes(1)
  })

  test('disable', () => {
    const click = jest.fn()
    const screen = render(<Cell label="限时奖励" value="六月阶梯奖励活动" arrow disabled onClick={click} />)
    fireEvent.click(screen.getByTestId('arrow'))
    fireEvent.click(screen.getByText('限时奖励'))
    fireEvent.click(screen.getByText('六月阶梯奖励活动'))
    expect(click).toHaveBeenCalledTimes(0)
  })

  test('suffix', () => {
    const screen = render(<Cell label="限时奖励" value="六月阶梯奖励活动" suffix="重选" arrow />)
    screen.getByText('限时奖励')
    screen.getByText('六月阶梯奖励活动')
    screen.getByText('重选')
    screen.getByTestId('arrow')
    expect(screen.container).toMatchSnapshot()
  })

  test('label = ReactNode', () => {
    const screen = render(<Cell label={<Text>自定义标题</Text>} value="六月阶梯奖励活动" suffix="重选" arrow />)
    screen.getByText('自定义标题')
    screen.getByText('六月阶梯奖励活动')
    screen.getByText('重选')
    screen.getByTestId('arrow')
    expect(screen.container).toMatchSnapshot()
  })


  test('value = ReactNode', () => {
    const screen = render(<Cell label="限时奖励" value={<Text>自定义内容</Text>} arrow />)
    screen.getByText('限时奖励')
    screen.getByText('自定义内容')
    screen.getByTestId('arrow')
    expect(screen.container).toMatchSnapshot()
  })


  test('suffix = ReactNode', () => {
    const screen = render(<Cell label="限时奖励" value="六月阶梯奖励活动" suffix={<Text>suffix</Text>} />)
    screen.getByText('限时奖励')
    screen.getByText('六月阶梯奖励活动')
    screen.getByText('suffix')
    expect(screen.container).toMatchSnapshot()
  })

  test('hide arrow', () => {
    const screen = render(<Cell label="限时奖励" value={<Text>自定义内容</Text>} />)
    screen.getByText('限时奖励')
    screen.getByText('自定义内容')
    expect(screen.queryByTestId('arrow')).toBeNull()
    expect(screen.container).toMatchSnapshot()
  })


  test('href', async () => {
    const screen = render(
    <BrowserRouter>
      <Cell label="限时奖励" value="前往查看" to="/home" />
    </BrowserRouter>)
    screen.getByText('限时奖励')
    screen.getByText('前往查看')
    screen.getByTestId('arrow')
    await waitFor(() => expect(screen.container.querySelector('a').href).toBe('http://localhost/home'))
    expect(screen.container).toMatchSnapshot()
  })

  test('href and hide arrow', async () => {
    const screen = render(
      <BrowserRouter>
        <Cell label="限时奖励" value="前往查看" to="/home" arrow={false} />
      </BrowserRouter>
    )
    screen.getByText('限时奖励')
    screen.getByText('前往查看')
    expect(screen.queryByTestId('arrow')).toBeNull()
    await waitFor(() => expect(screen.container.querySelector('a').href).toBe('http://localhost/home'))
    expect(screen.container).toMatchSnapshot()
  })

  test('href and disable', async () => {
    const click = jest.fn()
    const screen = render(
      <BrowserRouter>
        <Cell label="限时奖励" value="前往查看" to="/home" onClick={click} disabled />
      </BrowserRouter>
    )
    fireEvent.click(screen.getByText('限时奖励'))
    expect(click).toHaveBeenCalledTimes(0)
  })

  test('placeholder', () => {
    const screen = render(<Cell label="限时奖励" value="" placeholder="请选择奖励类型" arrow />)
    screen.getByText('限时奖励')
    screen.getByText('请选择奖励类型')
    screen.getByTestId('arrow')
    expect(screen.container).toMatchSnapshot()
  })

  test('placeholder And Value', () => {
    const screen = render(<Cell label="限时奖励" value="123" placeholder="请选择奖励类型" arrow />)
    screen.getByText('限时奖励')
    screen.getByText('123')
    expect(screen.queryByAltText('请选择奖励类型')).toBeNull()
    screen.getByTestId('arrow')
    expect(screen.container).toMatchSnapshot()
  })
})
