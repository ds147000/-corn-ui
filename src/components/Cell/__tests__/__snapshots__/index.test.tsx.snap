// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell basis 1`] = `
<div>
  <div
    class="corn-cell corn-f corn-ac"
  >
    <div
      class="corn-cell-label"
    >
      限时奖励
    </div>
    <div
      class="corn-cell-value"
    >
      六月阶梯奖励活动
    </div>
    <div
      class="corn-cell-suffix corn-f corn-ac"
    >
      <i
        class="corn-icon corn-i-big-right"
        data-testid="arrow"
      />
    </div>
  </div>
</div>
`;

exports[`Cell children 1`] = `
<div>
  <div
    class="corn-cell corn-f corn-ac"
  >
    <div
      class="corn-cell-label"
    >
      限时奖励
    </div>
    <div
      class="corn-cell-value"
    >
      <span
        class="taro-text"
      >
        六月阶梯奖励活动
      </span>
    </div>
    <div
      class="corn-cell-suffix corn-f corn-ac"
    >
      <i
        class="corn-icon corn-i-big-right"
        data-testid="arrow"
      />
    </div>
  </div>
</div>
`;

exports[`Cell hide arrow 1`] = `
<div>
  <div
    class="corn-cell corn-f corn-ac"
  >
    <div
      class="corn-cell-label"
    >
      限时奖励
    </div>
    <div
      class="corn-cell-value"
    >
      <span
        class="taro-text"
      >
        自定义内容
      </span>
    </div>
    <div
      class="corn-cell-suffix corn-f corn-ac"
    />
  </div>
</div>
`;

exports[`Cell href 1`] = `
<div>
  <a
    class="corn-link corn-link-normal corn-link-view corn-cell corn-f corn-ac"
    href="/home"
  >
    <div
      class="corn-cell-label"
    >
      限时奖励
    </div>
    <div
      class="corn-cell-value"
    >
      前往查看
    </div>
    <div
      class="corn-cell-suffix corn-f corn-ac"
    >
      <i
        class="corn-icon corn-i-big-right"
        data-testid="arrow"
      />
    </div>
  </a>
</div>
`;

exports[`Cell href and hide arrow 1`] = `
<div>
  <a
    class="corn-link corn-link-normal corn-link-view corn-cell corn-f corn-ac"
    href="/home"
  >
    <div
      class="corn-cell-label"
    >
      限时奖励
    </div>
    <div
      class="corn-cell-value"
    >
      前往查看
    </div>
    <div
      class="corn-cell-suffix corn-f corn-ac"
    />
  </a>
</div>
`;

exports[`Cell label = ReactNode 1`] = `
<div>
  <div
    class="corn-cell corn-f corn-ac"
  >
    <div
      class="corn-cell-label"
    >
      <span
        class="taro-text"
      >
        自定义标题
      </span>
    </div>
    <div
      class="corn-cell-value"
    >
      六月阶梯奖励活动
    </div>
    <div
      class="corn-cell-suffix corn-f corn-ac"
    >
      重选
      <i
        class="corn-icon corn-i-big-right"
        data-testid="arrow"
      />
    </div>
  </div>
</div>
`;

exports[`Cell placeholder 1`] = `
<div>
  <div
    class="corn-cell corn-f corn-ac"
  >
    <div
      class="corn-cell-label"
    >
      限时奖励
    </div>
    <div
      class="corn-cell-value"
    >
      
      <span
        class="taro-text corn-cell-placeholder"
      >
        请选择奖励类型
      </span>
    </div>
    <div
      class="corn-cell-suffix corn-f corn-ac"
    >
      <i
        class="corn-icon corn-i-big-right"
        data-testid="arrow"
      />
    </div>
  </div>
</div>
`;

exports[`Cell placeholder And Value 1`] = `
<div>
  <div
    class="corn-cell corn-f corn-ac"
  >
    <div
      class="corn-cell-label"
    >
      限时奖励
    </div>
    <div
      class="corn-cell-value"
    >
      123
    </div>
    <div
      class="corn-cell-suffix corn-f corn-ac"
    >
      <i
        class="corn-icon corn-i-big-right"
        data-testid="arrow"
      />
    </div>
  </div>
</div>
`;

exports[`Cell suffix = ReactNode 1`] = `
<div>
  <div
    class="corn-cell corn-f corn-ac"
  >
    <div
      class="corn-cell-label"
    >
      限时奖励
    </div>
    <div
      class="corn-cell-value"
    >
      六月阶梯奖励活动
    </div>
    <div
      class="corn-cell-suffix corn-f corn-ac"
    >
      <span
        class="taro-text"
      >
        suffix
      </span>
    </div>
  </div>
</div>
`;

exports[`Cell suffix 1`] = `
<div>
  <div
    class="corn-cell corn-f corn-ac"
  >
    <div
      class="corn-cell-label"
    >
      限时奖励
    </div>
    <div
      class="corn-cell-value"
    >
      六月阶梯奖励活动
    </div>
    <div
      class="corn-cell-suffix corn-f corn-ac"
    >
      重选
      <i
        class="corn-icon corn-i-big-right"
        data-testid="arrow"
      />
    </div>
  </div>
</div>
`;

exports[`Cell value = ReactNode 1`] = `
<div>
  <div
    class="corn-cell corn-f corn-ac"
  >
    <div
      class="corn-cell-label"
    >
      限时奖励
    </div>
    <div
      class="corn-cell-value"
    >
      <span
        class="taro-text"
      >
        自定义内容
      </span>
    </div>
    <div
      class="corn-cell-suffix corn-f corn-ac"
    >
      <i
        class="corn-icon corn-i-big-right"
        data-testid="arrow"
      />
    </div>
  </div>
</div>
`;
